File: trunk/EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_attack2_state.lua
class 'CPlayerAnimationAttack2State' (CState)
	function CPlayerAnimationAttack2State:__init(name) 
		CState.__init(self, name)
	end

	function CPlayerAnimationAttack2State:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("idle")
			_CCharacter:get_animation_model():blend_cycle( num, 0.1 )
		
			num = _CCharacter:get_animation_id("attack2")
			_CCharacter:get_animation_model():execute_action( num, 0.1 )
		end
	end
	
	function CPlayerAnimationAttack2State:Execute(_CCharacter)
	end
	
	function CPlayerAnimationAttack2State:OnExit(_CCharacter)
		num = _CCharacter:get_animation_id("idle")
		_CCharacter:get_animation_model():clear_cycle( num, 0.3 )
	end
	
	function CPlayerAnimationAttack2State:OnMessage(_CCharacter, _Msg)
		return false
	end
	
	function CPlayerAnimationAttack2State:__Finalize()
	end

File: trunk/EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationAttack2State.cpp
#include "PlayerAnimationAttack2State.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationAttack2State::CPlayerAnimationAttack2State( const std::string &_Name )
	: CState(_Name)
{

}

CPlayerAnimationAttack2State::~CPlayerAnimationAttack2State()
{

}

void CPlayerAnimationAttack2State::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->BlendCycle(l_iAnimID, 0.1f);

	l_iAnimID = _pCharacter->GetAnimationID("attack2");
	_pCharacter->GetAnimatedModel()->ExecuteAction(l_iAnimID, 0.1f);
}

void CPlayerAnimationAttack2State::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationAttack2State::OnExit( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("idle");
	_pCharacter->GetAnimatedModel()->ClearCycle(l_iAnimID, 0.1f);
}

bool CPlayerAnimationAttack2State::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}